src/Repository/AdherentChangeEmailTokenRepository.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace AppBundle\Repository;
+
+use AppBundle\Entity\Adherent;
+use AppBundle\Entity\AdherentChangeEmailToken;
+use Symfony\Bridge\Doctrine\RegistryInterface;
+
+class AdherentChangeEmailTokenRepository extends AbstractAdherentTokenRepository
+{
+    public function __construct(RegistryInterface $registry)
+    {
+        parent::__construct($registry, AdherentChangeEmailToken::class);
+    }
+
+    public function findLastUnusedByAdherent(Adherent $adherent): ?AdherentChangeEmailToken
+    {
+        return $this
+            ->createQueryBuilder('token')
+            ->where('token.adherentUuid = :uuid')
+            ->andWhere('token.usedAt IS NULL')
+            ->andWhere('token.expiredAt >= :date')
+            ->setParameters([
+                'uuid' => $adherent->getUuidAsString(),
+                'date' => new \DateTime(),
+            ])
+            ->orderBy('token.createdAt', 'DESC')
+            ->setMaxResults(1)
+            ->getQuery()
+            ->getOneOrNullResult()
+        ;
+    }
+
+    public function findLastUnusedByEmail(string $emailAddress): ?AdherentChangeEmailToken
```

Comment:
the queryBuilder in this method slightly differs from the one in `findLastUnusedByAdherent` method, should we refactor this and create a common queryBuilder with custom conditions/parameters in each method ?